### ISSUE TITLE
Revise the `Upgradeable wallet` reference design

### ISSUE THREAD

**Issue by GBKS** [BitcoinDesign/Guide#777]
This [page](https://bitcoin.design/guide/upgradeable-wallet/) has not been updated in a while. Content is slim and the visuals are minimal compared to the [Daily spending wallet](https://bitcoin.design/guide/daily-spending-wallet/) reference design. There are probably also more opportunities for cross-referencing relevant content now.

This issue is extracted from #717.

**Comment by GBKS** [comment]
Part of this revision can also be to expand the scope. The current page is only concerned with security. Another reason to upgrade may also be to take advantage of advanced features. For example, switching to Taproot allows for key rotation in multisig setups (via FORST), and multisig on LN. But Taproot is not widely adopted, so even if users upgrade, they may not be able to use features yet. How can upgrades to new standards be handled gracefully?

This came out of a discussion on Slack in the bech32m-adoption channel.

**Comment by Bosch-0** [comment]
This reference design always kind of confused me, I feel like this is more of a principle that should be used on any kind of bitcoin app (daily spending / savings etc.) rather than a type of product.

**Comment by GBKS** [comment]
Yes, not all reference designs need to be full products. That would make for a lot of repetition. They can also focus on individual user flows, features, etc. Like the [Sign in with bitcoin](https://github.com/BitcoinDesign/Guide/pull/885) PR.
